### PR TITLE
feat: allow snapshot warmup

### PR DESCRIPTION
**Pull request by littledivy**
Let embedders warmup the snapshot using a warmup script.

### CODE REVIEW

**Comment by mmastrac** [comment]
Can we start a deno branch to see what the changes look like there?

**Comment by littledivy** [comment]
Sure -- https://github.com/denoland/deno/pull/22386

File: core/extensions.rs [hunk 1]
```diff
@@ -581,6 +581,29 @@ pub struct Extension {
   pub enabled: bool,
 }
 
+impl Extension {
+  // Produces a new extension that is suitable for use during the warmup phase.
+  //
+  // JS sources are not included, and ops are include for external references only.
+  pub fn for_warmup(&self) -> Extension {
```
Comment (mmastrac) [inline core/extensions.rs:588]:
```suggestion
  pub(crate) fn for_warmup(&self) -> Extension {
```

**Review by mmastrac — APPROVED** [review]
LGTM. I'm happy we can do this without JsRuntime changes.

**Pull request merged**